### PR TITLE
libjpeg-turbo updates

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR updates the libjpeg-turbo package: it adds a new 'pic' variant, and allows `find_libraries` to find libraries for the static case.

### CODE REVIEW

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot fix style

**Review by becker33 — COMMENTED** [review]
Should this package have `conflicts("~shared~static")`? Does it build anything in that case?

**Comment by AlexanderRichert-NOAA** [comment]
Actually I think a named multi-choice variant called "libs" would be most consistent with other packages, I'll make the change in a moment...

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py [hunk 1]
```diff
@@ -58,9 +60,15 @@ class LibjpegTurbo(CMakePackage, AutotoolsPackage):
...
-    variant("static", default=True, description="Build static libs")
+    variant(
+        "libs",
+        default=("shared", "static"),
+        values=("shared", "static"),
+        multi=True,
+        description="Build shared libs, static libs, or both",
+    )
     variant("jpeg8", default=False, description="Emulate libjpeg v8 API/ABI")
+    variant("pic", default=True, description="Enable position independent code")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: Should we ~activate this one only when~ make this variant conditional on `spec.satisfies("libs=static")`?

Comment (AlexanderRichert-NOAA) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: I think so yes. Would it make sense to also have a conflict between ~pic and libs=shared to additionally avoid the `~pic libs=shared,static` case?

Comment (alalazo) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: No, that would be redundant. Just:
```suggestion
    variant("pic", default=True, description="Enable position independent code", when="libs=static")
```

**Comment by alalazo** [comment]
@spackbot fix style

**Review by alalazo — APPROVED** [review]

**Pull request merged**